src/ast/context.cpp

```diff
@@ -31,4 +31,11 @@ ASTContext::ASTContext() : ASTContext("", "")
 {
 }
 
+void ASTContext::clear()
```

Comment:
Doesn't reset `source_`?

Reply:
`source_` is constant. It is clearing all the diagnostics and nodes. I will add a comment that clarifies.

Reply:
Added.